### ISSUE TITLE
Straight lines in GeoJSON sometimes crash with normalized result is not a number

### ISSUE THREAD

**Issue by OmarShehata** [CesiumGS/cesium#8464]
The GeoJSON below crashes in CesiumJS:

```
{"type":"LineString","coordinates":[ [0.0, 0.0], [10.0, 0.0], [10.0, 5.0] ]}
```

![image](https://user-images.githubusercontent.com/1711126/70834242-d35ce780-1dc7-11ea-8556-4f0fcf0f5c5d.png)

_Visualized in [geojson.io](http://geojson.io/#map=6/3.426/0.747)_.

[Sandcastle link](https://sandcastle.cesium.com/index.html#c=xVFNawIxFPwrIRdXWJLdQy91lYKFQhF6sPRiPMTNaxuazZMku6Kl/71v/aiKP6A5BObNm2Ey6XRgnYUNBDZmHjZsCtG2jXjbz7JBvYdT9ElbD2EwHCnfkcbopOfYhhpId9Q8AT5H9I9/lHCoTfateNquQfF7xWdkMk/B+g/Fc8VrxGCs1wki0Qu2KESRM7qWOVuUt+COAFv+9CGk/NcYRVGekhzqE+coUWhjsjO+WNohNq+YHRH4ZJOFOBzxnFcxbR1MlGd0HmyzxpBYG1wmhEzQrF0fT67a+guSqGPsXSt5ElXGdsyaMT3m+scUZ7XTMRLz3jo3tzuqYFJJ2r+S9R1RHy8dBKe3/cpnOZkdhkKIShK8VSVEt9LhwvEX). Error produced:

```
DeveloperError: normalized result is not a number
Error
    at new DeveloperError (http://localhost:8080/Source/Workers/Check-da037458.js:40:23)
    at Function.Cartesian3.normalize (http://localhost:8080/Source/Workers/Cartesian2-2a723276.js:408:23)
    at Ellipsoid.geodeticSurfaceNormal (http://localhost:8080/Source/Workers/Cartesian2-2a723276.js:1800:31)
    at Ellipsoid.cartesianToCartographic (http://localhost:8080/Source/Workers/Cartesian2-2a723276.js:1890:26)
    at Object.GeometryPipeline.projectTo2D (http://localhost:8080/Source/Workers/GeometryPipeline-14f03628.js:980:40)
    at geometryPipeline (http://localhost:8080/Source/Workers/PrimitivePipeline-a698899f.js:261:63)
    at Object.PrimitivePipeline.combineGeometry (http://localhost:8080/Source/Workers/PrimitivePipeline-a698899f.js:367:30)
    at combineGeometry (http://localhost:8080/Source/Workers/combineGeometry.js:6:63)
    at callAndWrap (http://localhost:8080/Source/Workers/createTaskProcessorWorker.js:39:35)
    at http://localhost:8080/Source/Workers/createTaskProcessorWorker.js:84:34
```

But notice if you preturb the line just slightly, it renders fine:

```
{"type":"LineString","coordinates":[ [0.0, 0.0], [10.0, 0.0], [10.0, 5.0001] ]}
```

**Comment by dennisadams** [comment]
After some digging I think I've got it.  
It's an edge case in the generation of rhumb arcs in [PolylinePipeline.js](https://github.com/CesiumGS/cesium/blob/deaf3dfed7dec7262ecd11c05b28c0f0b659907e/Source/Core/PolylinePipeline.js).  

In [generateRhumbArc](https://github.com/CesiumGS/cesium/blob/deaf3dfed7dec7262ecd11c05b28c0f0b659907e/Source/Core/PolylinePipeline.js#L425), first the number of wanted points on each line segment is computed based on distance and granularity. This is used to create an empty positions array.  
Then [generateCartesianRhumbArc](https://github.com/CesiumGS/cesium/blob/deaf3dfed7dec7262ecd11c05b28c0f0b659907e/Source/Core/PolylinePipeline.js#L133) is called to compute the actual positions of the points, and before doing so it computes again the number of points on the line segment, to see how many positions it must generate.  
However, the second computation of the number of points is slightly different (see below). So in this edge case, there is one position "missing" in the positions array, which gradually turns into `undefined`s and then into destructive `NaN`s.  

Now for the details, with @OmarShehata's specific example in mind:  
First,  [generateRhumbArc](https://github.com/CesiumGS/cesium/blob/deaf3dfed7dec7262ecd11c05b28c0f0b659907e/Source/Core/PolylinePipeline.js#L425) computes the number of points on each segment:  
https://github.com/CesiumGS/cesium/blob/deaf3dfed7dec7262ecd11c05b28c0f0b659907e/Source/Core/PolylinePipeline.js#L463-L475
The first segment has 10 points and the second has 6. If you put a breakpoint inside [numberOfPointsRhumbLine](https://github.com/CesiumGS/cesium/blob/deaf3dfed7dec7262ecd11c05b28c0f0b659907e/Source/Core/PolylinePipeline.js#L24), you can see this 6 is the result of `Math.ceil(5.000000000000001)`.  
Then [generateCartesianRhumbArc](https://github.com/CesiumGS/cesium/blob/deaf3dfed7dec7262ecd11c05b28c0f0b659907e/Source/Core/PolylinePipeline.js#L488) is called for each segment, and starts by computing again the number of points:   
https://github.com/CesiumGS/cesium/blob/deaf3dfed7dec7262ecd11c05b28c0f0b659907e/Source/Core/PolylinePipeline.js#L143-L152
The first segment still has 10 points, but this time the second one has only 5 points.  Have another look at the breakpoint inside [numberOfPointsRhumbLine](https://github.com/CesiumGS/cesium/blob/deaf3dfed7dec7262ecd11c05b28c0f0b659907e/Source/Core/PolylinePipeline.js#L24) and you'll see this 5 is the result of `Math.ceil(4.999999999999999)`.  
So the 6th position is never generated and will remain empty in the positions array.   

Why is there a difference? In the first time, the `Cartesian3` end points are converted to `Cartographic`s before the computation. But in the second time, the `Cartesian3`s are scaled with `scaleToGeodeticSurface` before they are converted. This extra scaling makes a really tiny difference to some coordinates, but it turns out to be critical in this case.  

A simple solution that worked for me is to remove the scaling part (which is really redundant anyway), and instead to set the `height` of the converted `Cartographic`s to 0. (Notice that [numberOfPointsRhumbLine](https://github.com/CesiumGS/cesium/blob/deaf3dfed7dec7262ecd11c05b28c0f0b659907e/Source/Core/PolylinePipeline.js#L24) doesn't account for heights; perhaps it should, but that's another issue.)  
A better solution would obviously be to compute the number of points for each segment only once and pass it over to  [generateCartesianRhumbArc](https://github.com/CesiumGS/cesium/blob/deaf3dfed7dec7262ecd11c05b28c0f0b659907e/Source/Core/PolylinePipeline.js#L488).

**Comment by OmarShehata** [comment]
Thanks so much for digging into this @dennisadams ! It would be great if you can open a PR with that fix you suggested.

**Comment by dennisadams** [comment]
Great, I'll try to get it done by tomorrow.
